File: PS/algo/src/Leetcode/easy/BinaryPrefixDivisibleBy5.java
package Leetcode.easy;

import java.util.ArrayList;
import java.util.List;

public class BinaryPrefixDivisibleBy5 {
    public List<Boolean> prefixesDivBy5(int[] nums) {
        ArrayList<Boolean> answer = new ArrayList<>();
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum = sum * 2 + nums[i];
            if (sum % 5 == 0) {
                answer.add(true);
            } else {
                answer.add(false);
            }
            sum%=5;
        }

        return answer;
    }
}

File: src/main/kotlin/BinaryPrefixDivisibleBy5.kt
class BinaryPrefixDivisibleBy5 {
    fun prefixesDivBy5(nums: IntArray): List<Boolean> {
        var sum = 0
        val list = ArrayList<Boolean>()
        for (num in nums) {
            sum = sum * 2 +num
            list.add(sum % 5 == 0)
            sum%=5
        }
        return list
    }
}